test/apps/iodemo/ucx_wrapper.cc

```diff
@@ -445,12 +445,12 @@ void UcxContext::progress_io_message()
     } else {
         UcxConnection *conn = iter->second;
         if (!conn->is_established()) {
-            // tag-recv request can be completed before stream-recv callback has
-            // been invoked, go to another progress round and handle this io-msg
-            return;
-        }
-
-        if (conn->ucx_status() == UCS_OK) {
+            // tag-recv request can be completed before stream-recv callback
+            // has been invoked, cache this operation to handle it when
+            // connected and go to another progress round
+            conn->iomsg_recv_op_save(_iomsg_buffer,
```

Comment:
it means that before this PR we dropped the messages, and now we keep them?

Reply:
I don't think so. If we don't handle iomsg and don't complete and repost iomsg request, we don't drop it and should handle it on next iteration then connection is established.

Reply:
no, we just waiting for the connection being established and not processing new messages, because new tag_recv isn't posted.
so, waiting for the connection is being established blocks other connections to process iomsgs

Reply:
i see, thanks for the explanation!